streams/src/test/java/org/apache/kafka/streams/integration/RelationalSmokeTestIntegrationTest.java

```diff
@@ -0,0 +1,187 @@
+            return passed;
+        }
+    }
+
+    @Before
+    public void setUp() {
+        IntegrationTestUtils.cleanStateBeforeTest(CLUSTER, 2, RelationalSmokeTest.topics());
+    }
+
+    @Test
+    public void shouldWorkOnSingleNodeWithoutBounce() throws InterruptedException {
+        final String appId = "RelationalSmokeTestIntegration-" + UUID.randomUUID();
+
+        final RelationalSmokeTest.DataSet dataSet = RelationalSmokeTest.DataSet.generate(2, 2);
+        dataSet.produce(CLUSTER.bootstrapServers(), Duration.ZERO);
+
+        // take a nap
+        Thread.sleep(1000);
+
+        final boolean passed;
+        try (final KafkaStreams ignored = RelationalSmokeTest.App.startSync(CLUSTER.bootstrapServers(),
+                                                                            appId,
+                                                                            appId + "-0",
+                                                                            TestUtils.tempDirectory().getAbsolutePath())) {
+            passed = RelationalSmokeTest.App.verifySync(
+                CLUSTER.bootstrapServers(),
+                Instant.now().plus(Duration.ofMinutes(5))
+            );
+        }
+
+        Assert.assertTrue(passed);
+    }
+
+    @Test
+    public void shouldWorkWithRebalance() throws InterruptedException {
+        int bounces = 0;
+        int creates = 0;
+        final Instant start = Instant.now();
+        int numClientsCreated = 0;
+        final List<KafkaStreams> clients = new ArrayList<>();
+
+        final String appId = "RelationalSmokeTestIntegration-" + UUID.randomUUID();
+
+        final Driver driver = new Driver(
+            CLUSTER.bootstrapServers(),
+            1000,
+            10_000,
+            Duration.ofSeconds(20)
+        );
+        driver.start();
+        LOG.info("started driver");
+
+
+        // cycle out Streams instances a few times
+        while (driver.isAlive() && bounces < 3) {
+            // take a nap
+            Thread.sleep(1000);
+
+            // add a new client
+            final String id = appId + "-" + numClientsCreated++;
+            clients.add(
+                RelationalSmokeTest.App.startSync(
+                    CLUSTER.bootstrapServers(),
+                    appId,
+                    id,
+                    TestUtils.tempDirectory().getAbsolutePath()
+                )
+            );
+            creates++;
+            LOG.info("Create #{}", creates);
+
+            // let the oldest client die of "natural causes"
+            if (clients.size() >= 3) {
+                clients.remove(0).close(Duration.ZERO);
+                bounces++;
+                LOG.info("Bounce #{}", bounces);
+            }
+        }
+        try {
+            // wait for verification to finish
+            driver.join(IntegrationTestUtils.DEFAULT_TIMEOUT);
+        } finally {
+            // whether or not the assertions failed, tell all the streams instances to stop
```

Comment:
With a little distance, I think maybe we don’t need this integration test after all. It was useful in development, but I think I’ll just drop it. We’ll still have the unit test for the smoke test logic, but otherwise we’ll just rely on the system test itself. 

If we have trouble with the system test later on, we can always reintroduce this integration test at that point.